onnxruntime/core/providers/cpu/tensor/cast_op.cc

```diff
@@ -77,19 +90,38 @@ CastToString(const SrcType& input, std::string& output) {
       output = "INF";
     }
   } else {
-    // setprecision to 8 to match numpy default behavior
-    std::ostringstream convert;
-    convert << std::setprecision(8) << input;
-    output = convert.str();
+    // set precision to 8 to match numpy default behavior
+    constexpr const char* format = "%.8g";
+    const double value = static_cast<double>(input);
+
+    auto snprintf_result = std::snprintf(nullptr, 0, format, value);
+    ORT_ENFORCE(snprintf_result > 0, "Failed to determine required snprintf() buffer length.");
+
+    // buffer for string and trailing '\0'
```

Comment:
is there a maximum size that a float can possibly use when converted to a string that we could just use as the fixed buffer size to avoid the double call to snprintf?

Reply:
there probably is, but i'm not sure what it is. updated to make one call in the case where it does fit into a fixed buffer

---
In reply to: [569863895](https://github.com/microsoft/onnxruntime/pull/6509#discussion_r569863895) [](ancestors = 569863895)

Reply:
can we make the static buffer a lot bigger (I would use at least 256 bytes) so the chance of needing the fallback path is much lower? 

---
In reply to: [569930348](https://github.com/microsoft/onnxruntime/pull/6509#discussion_r569930348) [](ancestors = 569930348,569863895)